titiler/cmr/timeseries.py

```diff
@@ -0,0 +1,612 @@
+) -> List[CMRQueryParameters]:
+    """Timeseries query but without bbox as a parameter.
+
+    Needed this because FastAPI would expect bbox in the POST request body for
+    the /timeseries/statistics endpoint when using Depends(timeseries_query)
+    """
+    return timeseries_query(
+        concept_id, timeseries_params, minx=None, miny=None, maxx=None, maxy=None
+    )
+
+
+@define
+class TimeseriesExtension(FactoryExtension):
+    """Timeseries extension"""
+
+    def register(self, factory: Endpoints):
+        """Register timeseries endpoints to the MosaicTilerFactory"""
+        self.register_statistics(factory=factory)
+        self.register_tilejson(factory=factory)
+        self.register_images(factory=factory)
+
+        @factory.router.get(
+            "/timeseries",
+            response_model=TimeseriesCMRQueryParameters,
+            responses={
+                200: {
+                    "description": "Return the list of concept_id and datetime query parameters "
+                    "for a timeseries query"
+                }
+            },
+            tags=["Timeseries"],
+        )
+        def get_timeseries_parameters(
+            query=Depends(timeseries_query),
+        ):
+            return query
+
+    def register_statistics(self, factory: Endpoints):
+        """Register timeseries statistics endpoint"""
+
+        @factory.router.post(
+            "/timeseries/statistics",
+            response_model=TimeseriesStatisticsGeoJSON,
+            response_model_exclude_none=True,
+            response_class=GeoJSONResponse,
+            responses={
+                200: {
+                    "content": {"application/geo+json": {}},
+                    "description": "Return timeseries statistics for geojson features.",
+                }
+            },
+            tags=["Timeseries", "Statistics"],
+        )
+        async def timeseries_geojson_statistics(
+            request: Request,
+            geojson: Annotated[
+                Union[FeatureCollection, Feature],
+                Body(description="GeoJSON Feature or FeatureCollection.", embed=False),
+            ],
+            query=Depends(timeseries_query_no_bbox),
+            coord_crs=Depends(CoordCRSParams),
+            dst_crs=Depends(DstCRSParams),
+            rasterio_params=Depends(factory.rasterio_dependency),
+            zarr_params=Depends(factory.zarr_dependency),
+            reader_params=Depends(factory.reader_dependency),
+            post_process=Depends(factory.process_dependency),
+            stats_params=Depends(factory.stats_dependency),
+            histogram_params=Depends(factory.histogram_dependency),
+            image_params=Depends(factory.img_part_dependency),
+        ):
+            urls = build_request_urls(
+                base_url=str(factory.url_for(request, "geojson_statistics")),
+                request=request,
+                param_list=query,
+            )
```

Comment:
Just to make sure I understand how this works: the timeseries params are generated as part of the dependency `timeseries_query_no_bbox` - which generates a query object that is list of objects each containing a concept id and datetime. Build request urls returns a list of urls, each of which is a statistics endpoint, and then in the line below a request is made to each of those statistics endpoints. Is that understanding correct?

Reply:
@abarciauskas-bgse yes that is correct! The code in the `/timeseries` endpoints just converts the timeseries query parameters into a list of `datetime` parameters, then a fires off a lower-level request for each unique `datetime` parameter. Then there is a little bit of code to organize the results (e.g. list of PNGs to a GIF, list of statistics geojson output into a single geojson).